javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -76,10 +74,20 @@ predicate oneshotClosure(InvokeExpr call) {
   call.getCallee().getUnderlyingValue() instanceof ImmediatelyInvokedFunctionExpr
 }
 
+predicate alwaysThrows(Function f) {
```

Comment:
We need another performance evaluation for this one (I should have said that up front) since `BasicBlock` reasoning is prone to being expensive. It should be sufficient to check `183558d@62f5891` (your old HEAD) against your current HEAD for big-apps.slugs.

Reply:
So after introducing `alwaysThrows` the query is ... faster.  
https://git.semmle.com/erik/dist-compare-reports/tree/domitian.ti.semmle.com_1570544115413

Reply:
Did you run that with the same set of queries?
The previous dist-compare report in this PR also used Xss.ql:

```
options: --queries /home/erik/code/target/intree/profiling-dist-javascript/queries/semmlecode-javascript-queries/Security/CWE-079/Xss.ql /home/erik/code/target/intree/profiling-dist-javascript/queries/semmlecode-javascript-queries/Statements/UseOfReturnlessFunction.ql` for `183558d@62f5891
```
 which appears to have been reused in the dist-compare report you linked above :)

(solution - delete the `projects/*/revision-183558d@0933235`-directories, and run again with both Xss.ql and UseOfReturnlessFunction.ql included as a queries)

Reply:
Yeah, that was better.   
Performance is unchanged. 
https://git.semmle.com/erik/dist-compare-reports/tree/domitian.ti.semmle.com_1570605405712